src/Servers/Kestrel/Core/src/KestrelConfigurationLoader.cs

```diff
@@ -293,6 +293,8 @@ public void Load()
                     httpsOptions.ServerCertificate = LoadCertificate(endpoint.Certificate, endpoint.Name)
                         ?? httpsOptions.ServerCertificate;
 
+                    httpsOptions.ClientCertificateMode = ConfigurationReader.ClientCertificateMode ?? httpsOptions.ClientCertificateMode;
```

Comment:
Organization: This was inserted between to related code sections dealing with server certs. Move it up above line 293.